pyrsa/util/indicator.py

```diff
@@ -0,0 +1,74 @@
+"""
+Collection of different n_uniqueinds of indicator Matrices
+    identity: One column per unique element in vector
+    identity_pos: One column per unique non-zero element
+    allpairs:     All n_unique*(n_unique-1)/2 pairwise contrasts
+"""
+import numpy as np
+
+
+def identity(index_vector):
+    """ Indicator matriindicator_matrix with one
+        column per unique element in vector
+        Args:
+            index_vector (numpy.ndarray): n_row vector to
+            code - discrete values (one dimensional)
+        Returns:
+            indicator_matrix (numpy.ndarray): n_row indicator_matrix
+                n_values indicator matriindicator_matrix
+    """
+    c_unique = np.unique(index_vector)
+    n_unique = c_unique.size
+    rows = np.size(index_vector)
+    indicator_matrix = np.zeros((rows, n_unique))
+    for i in np.arange(n_unique):
+        indicator_matrix[index_vector == c_unique[i], i] = 1
+    return indicator_matrix
+
+
+def identity_pos(index_vector):
+    """ Indicator matriindicator_matrix with one column
+        per unique positive element in vector
+        Args:
+            c (numpy.ndarray): n_row vector to code -
+                               discrete values (one dimensional)
+        Returns:
+            indicator_matrix (numpy.ndarray): n_row indicator_matrix
+                n_values indicator matriindicator_matrix
+    """
+    c_unique = np.unique(index_vector)
+    n_unique = c_unique.size
+    rows = np.size(index_vector)
+    c_unique = c_unique[c_unique > 0]
+    n_unique = c_unique.size
+    indicator_matrix = np.zeros((rows, n_unique))
+    for i in range(n_unique):
+        indicator_matrix[index_vector == c_unique[i], i] = 1
+    return indicator_matrix
```

Comment:
On first look it seems like `identity` and `identity_pos` are quite similar, perhaps  `identity_pos` could call `identity` with an flag `positive=True`? It then checks for this flag and does the `cc[cc>0]` step. This solves the code replication warning.

Reply:
Thanks for the note! Will take a look and change in the next few pushes.

Reply:
fixed and pushed.